framework-plugins/lisk-framework-http-api-plugin/src/controllers/delegates.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { Request, Response, NextFunction } from 'express';
+import { validator, LiskValidationError } from '@liskhq/lisk-validator';
+import { BaseChannel, PluginCodec } from 'lisk-framework';
+import { paginateList } from '../utils';
+
+const getDelegatesQuerySchema = {
+	type: 'object',
+	properties: {
+		limit: {
+			type: 'string',
+			format: 'uint32',
+			description: 'Number of delegates to be returned',
+		},
+		offset: {
+			type: 'string',
+			format: 'uint32',
+			description: 'Offset to get delegates after a specific length in a delegates list',
+		},
+	},
+};
+
+export const getDelegates = (channel: BaseChannel, codec: PluginCodec) => async (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+): Promise<void> => {
+	const errors = validator.validate(getDelegatesQuerySchema, req.query);
+	// 400 - Malformed query or parameters
+	if (errors.length) {
+		res.status(400).send({
+			errors: [{ message: new LiskValidationError([...errors]).message }],
+		});
+		return;
+	}
+
+	const { limit = 100, offset = 0 } = req.query;
+	try {
+		const encodedDelegates: string[] = await channel.invoke('app:getAllDelegates');
+		const decodedDelegates = encodedDelegates.map(delegate => codec.decodeAccount(delegate));
+
+		res.status(200).send({
```

Comment:
Always use `res.json` unless have clear need of `res.send`. The difference is `res.send` tries to detect `Content-Type` header value from the parameter. And based on different type of parameter its value would be different. So be rather explicit to mention that your are sending JSON which means `Content-Type: application/json`

Also `res.send` tries to detect and set `Content-Length` header and few more things.